In [3]:
import numpy as np
import pandas as pd
import yfinance as yf

In [4]:
data = yf.download("AAPL", start="2026-01-01", end="2026-08-01")
print(data.shape)

C:\Users\fanis\AppData\Local\Temp\ipykernel_14008\1896517990.py:1: FutureWarning: YF.download() has changed argument auto_adjust default to True
  data = yf.download("AAPL", start="2026-01-01", end="2026-08-01")
[*********************100%***********************]  1 of 1 completed

(145, 5)


In [6]:
print(data.columns)
print(data.head())

MultiIndex([( 'Close', 'AAPL'),
            (  'High', 'AAPL'),
            (   'Low', 'AAPL'),
            (  'Open', 'AAPL'),
            ('Volume', 'AAPL')],
           names=['Price', 'Ticker'])
Price            Close        High         Low        Open    Volume
Ticker            AAPL        AAPL        AAPL        AAPL      AAPL
Date                                                                
2026-01-02  270.274323  277.085768  268.269769  271.520929  37838100
2026-01-05  266.534515  270.772978  265.417561  269.905345  45647200
2026-01-06  261.647797  266.823711  261.408458  266.275216  52352100
2026-01-07  259.623291  262.964203  259.104714  262.485526  48309800
2026-01-08  258.336853  258.586174  255.005908  256.322317  50419300


In [9]:
closes=data["Close"]
daily_change=closes.diff()
print(daily_change.head())
print(daily_change.describe())

Ticker          AAPL
Date                
2026-01-02       NaN
2026-01-05 -3.739807
2026-01-06 -4.886719
2026-01-07 -2.024506
2026-01-08 -1.286438
Ticker        AAPL
count   144.000000
mean      0.266455
std       5.138359
min     -24.498840
25%      -2.181107
50%       0.309273
75%       2.820061
max      14.237701


What data.shape of (145, 5) means, using the word "matrix": this is a 145 by 5 matrix, 145 rows for trading days, 5 columns for price fields.

What .diff() calculated and why the first row was NaN: .diff() computes the difference bettween the close day and the previous close day. NAN was the first point data and no previous day was in the data to subtract so NAN.

Why your mean prediction was wrong, and what that teaches you about judging data from a small sample: Mean was about 0.266455, however since is predicting the whole 7 months period is +ve. Compared to the first week which was negative it was a bad week for the stock. This implies small data set is not a good estimate for the whole picture